### DISCUSSION TITLE
Does this work on amd or other ai card or only with Cuda nvidia card?

### DISCUSSION THREAD

**Discussion by lollo1252** [nagadomi/nunif#247] [Q&A] [answered]
I am have not the money to upgrade my 3060 laptop with a 2000 euro card like the 4090, the question is can I get similar performance with specific ai card, something like Google coral but more powerfull, there is out there some of this card that support Cuda or pythorch and can work with this project?

**Comment by nagadomi** [comment, marked as answer]
This project is working with PyTorch.
PyTorch supports ROCm(AMD) on Linux, so this can work with AMD cards on Linux or WSL. https://github.com/nagadomi/nunif/discussions/166
macOS/MPS is already supported. 
Intel GPU is not yet supported, but it will be possible when PyTorch for xpu is officially released. https://github.com/nagadomi/nunif/discussions/232

Regarding performance, I don't know about the performance of cards I don't have. I am not sure that there is anything faster and low cost than RTX3060 Mobile.

**Reply by nagadomi** [reply]
Also, this project is 2-3x faster on Linux than on Windows. https://github.com/nagadomi/nunif/issues/77
If you are using Windows, you can boot and run free Ubuntu from a $10 USB drive. 
(Be careful not to destroy the Windows boot/data disk.)

**Reply by lollo1252** [reply]
Il try installing ubuntu on my 2nd ssd, but I like more arch or fedora If I need to be honest since arch run way faster than all the debian base distro I know about

**Reply by lollo1252** [reply]
> This project is working with PyTorch.
> PyTorch supports ROCm(AMD) on Linux, so this can work with AMD cards on Linux or WSL. https://github.com/nagadomi/nunif/discussions/166
> macOS/MPS is already supported. 
> Intel GPU is not yet supported, but it will be possible when PyTorch for xpu is officially released. https://github.com/nagadomi/nunif/discussions/232
> 
> Regarding performance, I don't know about the performance of cards I don't have. I am not sure that there is anything faster and low cost than RTX3060 Mobile.
> 
> 

My 3060 mobile is in a laptop, I want to build a pc with amd card like the rx7950xtx for the sweet 24gb vram that would be awesome for minecraft vr and other vr titles in comparison of 6gb of the 3060, but the 3060 have rt core or tensor or whatever they are called that work pretty good for ai, so a 4090 would be the best, but it cost double, at this point spend 1000 euros for the gaming gpu and the other 1000 for a specific ai accelerator card that is supported would probably give me better performance i hope

**Reply by nagadomi** [reply]
If you are familiar with Linux, use your preferred distribution. this repo should work with Python >= 3.10 and any linux.

Regarding AI Accelerator card, I personally think NVIDIA card is the only choice.
Because developers are less likely to support devices they do not use themselves. And most AI/ML people use NVIDIA card.

**Reply by beosliege** [reply]
EDIT: I just realized there was already a more detailed link before to my original post:
https://github.com/nagadomi/nunif/discussions/166


https://rocm.docs.amd.com/projects/radeon/en/latest/docs/install/wsl/howto_wsl.html

I use AMD Radeon RX 7900 XTX GPU.

I used to use a dedicated Ubuntu Linux install just for running iw3, but I got tired of dual-booting. WSL (windows subsystem for Linux) can be installed on Windows 10 or 11 and then you follow the above instructions VERY carefully to install ROCm and pytorch within the WSL. Make sure you verify each step along the way as the instructions indicate so that if something goes wrong you know where to troubleshoot.

After that you can always launch a WSL command prompt in Windows and then set up this iw3 python project. I run iw3 all the time like this and it works wonderfully. Currently converting a 4k video to 3D using about as high quality settings as one could specify at about 2 FPS (this is VERY high quality results, worth the time to wait for). On this card I am able to undervolt to 93% while increasing the power by 15% in the AMD Adrenaline software, and the resulting performance is very nice, even on air cooling. Quite happy with this setup, and looking forward to AMD continuing to support more and more AI / ML as they bring ROCm to maturity.

I think eventually we'll have this support natively in Windows without WSL, but for now it's so nice to have WSL so I don't have to dual-boot!
